server.js

```diff
@@ -7545,6 +7545,71 @@ camp.route(/^\/maven-metadata\/v\/(https?)\/(.+\.xml)\.(svg|png|gif|jpg|json)$/,
     });
 }));
 
+// badge-json url
+camp.route(/^\/json\/(https?.+)\.(svg|png|gif|jpg|json)$/,
```

Comment:
As I mentioned in https://github.com/badges/shields/pull/1525#issuecomment-370190084 let's make two or maybe three ways of leveraging the JSON functionality.

I'd prefer we add support for three endpoints:

- `/badge/dynamic/json-badge-data/https/example.com/info.json.svg` (as an alternative to /json`)
- `/badge/dynamic/json-badge-data.svg?url=https%2Fexample.com%2Finfo.json`
- `/badge/static.svg?badgeData=...`

These support the more efficient static route for servers capable of building a redirect. That route can also be used in a dynamic web page, which is really cool! Simultaneously they make the "static" and "dynamic" distinctions very clear.

Reply:
Good idea!
<details>
<summary>sidenote</summary>

I'm not much of a fan of the `https/example.com` endpoints,
just feels strange without the `https://`:
`/badge/dynamic/json-badge-data/https://example.com/path/to.json.svg`
`/badge/dynamic/json-badge-data/https/example.com/path/to.json.svg`
but we have a few that way already so for the sake of consistency it's probably best to continue that pattern.
</details>

Reply:
Yea… I find it a little weird too. Though, I think a colon in a path is supposed to be URL encoded. Maybe we could just make the scheme optional and default to `https`? That way it would disappear in most cases…